test/powershell/Language/Parser/Conversions.Tests.ps1

```diff
@@ -20,4 +20,41 @@ Describe 'conversion syntax' -Tags "CI" {
         # This test relies on the fact that there are overloads (at least 2) for ToString method.
         ([System.Management.Automation.ActionPreference]"Stop").ToString() | Should Be "Stop"
     }
+
+    Context "Cast object[] to more narrow generic collection" {
+        BeforeAll {
+            $testCases = @(
+                @{ Command = {$result = [Collections.Generic.List[int]]@(1)};      CollectionType = 'List`1'; ElementType = "Int32";  Count = 1; Elements = @(1) }
+                @{ Command = {$result = [Collections.Generic.List[int]]@(1,2)};    CollectionType = 'List`1'; ElementType = "Int32";  Count = 2; Elements = @(1,2) }
+                @{ Command = {$result = [Collections.Generic.List[int]]"4"};       CollectionType = 'List`1'; ElementType = "Int32";  Count = 1; Elements = @(4) }
+                @{ Command = {$result = [Collections.Generic.List[string]]@(1)};   CollectionType = 'List`1'; ElementType = "String"; Count = 1; Elements = @("1") }
+                @{ Command = {$result = [Collections.Generic.List[string]]@(1,2)}; CollectionType = 'List`1'; ElementType = "String"; Count = 2; Elements = @("1","2") }
+                @{ Command = {$result = [Collections.Generic.List[string]]1};      CollectionType = 'List`1'; ElementType = "String"; Count = 1; Elements = @("1") }
+
+                @{ Command = {$result = [System.Collections.ObjectModel.Collection[int]]@(1)};   CollectionType = 'Collection`1'; ElementType = "Int32"; Count = 1; Elements = @(1) }
+                @{ Command = {$result = [System.Collections.ObjectModel.Collection[int]]@(1,2)}; CollectionType = 'Collection`1'; ElementType = "Int32"; Count = 2; Elements = @(1,2) }
+                @{ Command = {$result = [System.Collections.ObjectModel.Collection[int]]"4"};    CollectionType = 'Collection`1'; ElementType = "Int32"; Count = 1; Elements = @(4) }

```

Comment:
I added one more set of tests for [List[FileInfo]] cast from string or array of strings. If you think more tests is needed in this area, feel free to open an issue.